source/loader/layers/sanitizer/asan_options.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+ * See LICENSE.TXT
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ * @file asan_options.cpp
+ *
+ */
+
+#include "asan_options.hpp"
+#include "ur_sanitizer_layer.hpp"
+
+#include <algorithm>
+#include <cstring>
+#include <stdexcept>
+
+namespace ur_sanitizer_layer {
+
+AsanOptions::AsanOptions() {
+    auto OptionsEnvMap = getenv_to_map("UR_LAYER_ASAN_OPTIONS");
```

Comment:
this will conflict with: https://github.com/oneapi-src/unified-runtime/pull/2054/files that I just merged. Please rebase.

Reply:
Done